### PR TITLE
fix issue_106: added a design diagram for this project

### PR DESCRIPTION
**Pull request by ahrtr**
fix https://github.com/mattn/goreman/issues/106

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,15 @@ See `_example` directory
...
 MIT
 
+## Design
+
+The main goroutine is driven by the following two kinds of events after it starts, and then take proper action against the managed processes.
+
+1. It receives a signal, which could be one of SIGINT,SIGTERM and SIGHUP;
+2. It receives a RPC call, which is triggered by the command "goreman run COMMAND [PROCESS...]".
+
+![design](images/design.png)
```
Comment (mattn) [inline README.md:33]: Missing image file?

Comment (ahrtr) [reply README.md:33]: The reason is that the domain name "raw.githubusercontent.com" couldn't be resolved somehow, and it works after adding the following entry into my /etc/hosts,

```
151.101.76.133 raw.githubusercontent.com
```

**Comment by mattn** [comment]
LGTM.

**Comment by mattn** [comment]
Thank you.

**Pull request merged**